src/App.jsx

```diff
@@ -0,0 +1,88 @@
+import './App.css';
+import {ProductList} from "./components/Product/ProductList";
+import {ProductPage}from "./components/Product/ProductPage";
+import {WishList} from "./components/Wishlist/WishList";
+import {CartList} from "./components/Cart/CartList";
+import {Address} from "./components/Address/Address";
+import {Home} from "./components/Home/Home";
+import {Login} from "./components/Home/Login";
+import {Signup} from "./components/Home/Signup";
+import { useTheme } from "./context/theme-context";
+import {useCart} from "./context/cart-context";
+import {useAxios} from "./useAxios";
+import {ReactComponent as DarkTheme} from "./asssets/dark-theme-white.svg";
+import {ReactComponent as LightTheme} from "./asssets/dark-theme.svg";
+import {ReactComponent as CartSvg} from "./asssets/cart.svg";
+import {ReactComponent as WishlistSvg} from "./asssets/wishlist.svg";
+import {Routes,Route,Link} from "react-router-dom";
+import {PrivateRoute} from "./PrivateRoute";
+import { useEffect, useState } from 'react';
+import {ReactComponent as TruckLogo} from './asssets/truck.svg';
+const selectedTheme={
+  "light":{
+    bg:"white",
+    color:"black"
+  },
+  "dark":{
+    bg:"black",
+    color:"white"
+  }
+}
+function App() {
+  const {theme,toggleTheme} = useTheme();
+  const {value:{cart,wishlist},showToast,setShowToast,textToast}=useCart();
+  const [cartItemsCount,setCartItemsCount]= useState(0);
+  useEffect(()=>{
+    setCartItemsCount(cart.reduce(((acc,curr)=>acc+parseInt(curr.quantity)),0));
+  },[cart])
+  console.log({cartItemsCount});
+  const {loading,error}="";
+  console.log(textToast);
+  if(showToast===true){
+    setTimeout(() => {
+      setShowToast(false);
+    }, 1000);
+  }
+  return (
+    <div style={{backgroundColor:selectedTheme[theme].bg,color:selectedTheme[theme].color,minHeight:"100vh"}}>
+      <nav className="nav" style={{backgroundColor:selectedTheme[theme].bg,color:selectedTheme[theme].color}} >
```

Comment:
You can try to extract out these components into separate files, maybe also  have another component that will list all the routes, and then just have that component here in the app.js file along with the navbar component